test/unit/org/apache/cassandra/metrics/CounterMutationMetricsTest.java

```diff
@@ -0,0 +1,174 @@
+    private static final String CF2 = "Counter2";
+
+    @BeforeClass
+    public static void defineSchema() throws ConfigurationException
+    {
+        SchemaLoader.prepareServer();
+        SchemaLoader.createKeyspace(KEYSPACE1,
+                                    KeyspaceParams.simple(1),
+                                    SchemaLoader.counterCFMD(KEYSPACE1, CF1),
+                                    SchemaLoader.counterCFMD(KEYSPACE1, CF2));
+    }
+
+    @Test
+    public void testLocksDuplicate()
+    {
+        AssertMetrics metrics = new AssertMetrics();
+
+        ColumnFamilyStore cfsOne = Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF1);
+        ColumnFamilyStore cfsTwo = Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF2);
+
+        cfsOne.truncateBlocking();
+        cfsTwo.truncateBlocking();
+
+        Mutation.PartitionUpdateCollector batch = new Mutation.PartitionUpdateCollector(KEYSPACE1, Util.dk("key1"));
+        batch.add(new RowUpdateBuilder(cfsOne.metadata(), 5, "key1")
+                  .clustering("cc")
+                  .add("val", 1L)
+                  .add("val2", -1L)
+                  .build().getPartitionUpdate(cfsOne.metadata()));
+
+        batch.add(new RowUpdateBuilder(cfsTwo.metadata(), 5, "key1")
+                  .clustering("cc")
+                  .add("val", 2L)
+                  .add("val2", -2L)
+                  .build().getPartitionUpdate(cfsTwo.metadata()));
+
+        new CounterMutation(batch.build(), ConsistencyLevel.ONE).apply();
+
+        metrics.assertLockTimeout(0);
+        metrics.assertLocksPerUpdate(1);
+    }
+
+    @Test
+    public void testLockTimeout() throws Throwable
+    {
+        AssertMetrics metrics = new AssertMetrics(11L);
+
+        ColumnFamilyStore cfs = Keyspace.open(KEYSPACE1).getColumnFamilyStore(CF1);
+        cfs.truncateBlocking();
+
+        // Obtain lock
+        new CounterMutation(
+        new RowUpdateBuilder(cfs.metadata(), 5, "key1").clustering("cc")
+                                                       .add("val", 1L)
+                                                       .build(),
+        ConsistencyLevel.ONE).grabCounterLocks(Keyspace.open(KEYSPACE1), new ArrayList<>());
+
+        metrics.assertLockTimeout(0);
+
+        // Try to mutate
+        CompletableFuture.runAsync(() ->
+                                   {
+                                       try
+                                       {
+                                           new CounterMutation(
+                                           new RowUpdateBuilder(cfs.metadata(), 5, "key1")
+                                           .clustering("cc")
+                                           .add("val", 1L)
+                                           .add("val2", -1L)
+                                           .build(),
+                                           ConsistencyLevel.ONE).apply();
+                                       }
```

Comment:
assert false after mutation application? can't hurt, and we no longer need to rely on lock timeout metric to fail the test.

The point is that now when the test fails we don't know if the test scenario was executed (and metric bumping is broken), or perhaps the application just succeeded (which means that our assumption about test scenario failed). tiny bit, but helps debugging